src/java/org/apache/cassandra/index/sai/disk/v2/V2VectorIndexSearcher.java

```diff
@@ -212,6 +220,19 @@ private BitsOrPostingList bitsOrPostingListForKeyRange(QueryContext context, Abs
                 return new BitsOrPostingList(new ArrayPostingList(postings.toIntArray()));
             }
 
+            if (graph instanceof CassandraDiskAnn && ((CassandraDiskAnn) graph).getCompressedVectors() != null)
```

Comment:
Minor improvement: add getCompressedVectors to the interface and have it default to null so we dn't have to do the instanceof dance